src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpPreferIsExactSpellingIsTrueForKnownApis.Fixer.cs

```diff
@@ -0,0 +1,164 @@
+            if (dllImportAttribute is null)
+            {
+                return;
+            }
+
+            var dllName = dllImportAttribute.ConstructorArguments[0].Value.ToString();
+            string title = MicrosoftNetCoreAnalyzersResources.PreferIsExactSpellingIsTrueForKnownApisTitle;
+
+            var editor = await DocumentEditor.CreateAsync(context.Document, context.CancellationToken).ConfigureAwait(false);
+            if (!PreferIsExactSpellingIsTrueForKnownApisAnalyzer.KnownApis.Value.TryGetValue(dllName, out var methods))
+            {
+                return;
+            }
+
+            var actualName = methodSymbol.GetActualExternName(dllImportAttribute);
+            if (methods.Contains(actualName))
+            {
+                if (actualName.EndsWith("W", StringComparison.OrdinalIgnoreCase))
+                {
+                    context.RegisterCodeFix(new PreferIsExactSpellingChangeAction(title,
+                                                     async ct => await ApplixFixAndGetChangedDocument(editor, AddExactSpelling(context.Document, methodDeclaration, editor, ct)).ConfigureAwait(false),
+                                                     equivalenceKey: title),
+                                    context.Diagnostics);
+                    return;
+                }
+                if (actualName.EndsWith("A", StringComparison.OrdinalIgnoreCase))
+                {
+                    context.RegisterCodeFix(new PreferIsExactSpellingChangeAction(title,
+                                                     async ct => await ApplixFixAndGetChangedDocument(editor, AddExactSpelling(context.Document, methodDeclaration, editor, ct)).ConfigureAwait(false),
+                                                     equivalenceKey: title),
+                                    context.Diagnostics);
+                    return;
+                }
+            }
+            if (methods.Contains(actualName + "A"))
+            {
+                context.RegisterCodeFix(new PreferIsExactSpellingChangeAction(title,
+                                                     async ct => await ApplixFixAndGetChangedDocument(editor, AddASuffix(context.Document, methodDeclaration, actualName, editor, ct)).ConfigureAwait(false),
+                                                     equivalenceKey: title),
+                                    context.Diagnostics);
+            }
+        }
+
+        public static async Task AddExactSpelling(Document document, MethodDeclarationSyntax methodDeclaration, DocumentEditor editor, CancellationToken cancellationToken)
+        {
+            SyntaxGenerator generator = editor.Generator;
+            SemanticModel model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var dllImportSyntax = methodDeclaration.AttributeLists.First(x => x.Attributes.Any(y => y.Name.ToString().Equals("DllImport", StringComparison.Ordinal)));
```

Comment:
I don't know about that, we prefer to check the symbols  rather than just string names. If you add a test where you use the long form `DllImportAttribute` it should fail.